### PR TITLE
Add aten::_foreach_norm

### PR DESCRIPTION
**Pull request by fengyuan14**

### CODE REVIEW

File: src/aten/sycl/ForeachReduceKernels.cpp [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    if (n % kILP == 0 && (chunk_size & kILP) == 0 && is_aligned(x)) {
+      for (int64_t i_start = item_idx;
+           i_start * kILP < n && i_start * kILP < chunk_size;
+           i_start += item_range) {
+        // load
+        load_store(r_x, x, 0, i_start);
+#pragma unroll
+        for (int ii = 0; ii < kILP; ii++) {
+          opmath_t next = static_cast<opmath_t>(r_x[ii]);
+          vals[ii] += NormType == 1
```
[the remark this answers is not in the document]
Comment (fengyuan14) [reply src/aten/sycl/ForeachReduceKernels.cpp:56]: Let me rebase to align with CUDA.

File: src/ATen/native/xpu/ForeachReduceOp.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    const Scalar& ord,
+    c10::optional<ScalarType> dtype) {
+  const auto p = [&]() -> double {
+    if (ord.isIntegral(false)) {
+      return ord.to<int64_t>();
+    } else if (ord.isFloatingPoint()) {
+      return ord.to<double>();
+    } else {
+      TORCH_CHECK(
+          false, "foreach_tensor_norm_cuda expects ord to be integer or float");
```
Comment (xytintel) [inline src/ATen/native/xpu/ForeachReduceOp.cpp:53]: Pls remove `cuda`

File: src/ATen/native/xpu/ForeachReduceOp.cpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  }();
+  at::native::check_foreach_api_restrictions(tensors);
+  const bool has_int_or_complex =
+      std::any_of(tensors.begin(), tensors.end(), [](const auto& t) {
+        const auto scalar_type = t.scalar_type();
+        return at::isIntegralType(scalar_type, /*includeBool*/ true) ||
+            at::isComplexType(scalar_type);
+      });
+  if (!at::native::can_use_fast_route(tensors) || has_int_or_complex ||
+      !(p == static_cast<double>(1) || p == static_cast<double>(2))) {
```
Comment (xytintel) [inline src/ATen/native/xpu/ForeachReduceOp.cpp:64]: Please align with the latest condition judgments:  `|| p == std::numeric_limits<double>::infinity())`

File: src/ATen/native/xpu/ForeachReduceOp.cpp (hunk 2 above)
Comment (fengyuan14) [reply src/ATen/native/xpu/ForeachReduceOp.cpp:53]: WIll fix

**Review by xytintel — APPROVED** [review]

**Pull request merged**